Fix issue with non-ascii character in shell script text

hell0again opened the pull request:
I got `UnicodeDecodeError` when using non-ascii character in shell script.

```
UnicodeDecodeError: 'utf8' codec can't decode byte 0xe3 in position 0: unexpected end of data
```

## How to reproduce

I added some non-ascii test. This test can can reproduce the error on `v1.0.11` or `v1.1.0`(latest), but not on `v1.0.10`. So is seems `v1.0.11` is the first version to reproduce.

https://github.com/ktdreyer/jenkins-job-wrecker/compare/v1.0.10...v1.0.11

I think this error occurred by following reasons.

- `xml.etree.ElementTree` parses xml text node as type `<unicode>`, when a xml text node contains non-ascii character.

- In Shell builder, shell script text was converted to type `str` by calling `str(shell_element.text)`.   https://github.com/ktdreyer/jenkins-job-wrecker/blob/master/jenkins_job_wrecker/builders.py#L89  
  This convert runs anytime even if shell script text was parsed as type `<unicode>` by containing non-ascii character. Note that we can reproduce `UnicodeDecodeError` by following.

```
#!/usr/bin/env python2.7
# -*- coding: utf-8 -*-
import sys
import yaml
reload(sys)
sys.setdefaultencoding('utf-8')
yaml.add_representer(str, lambda dumper, data: dumper.represent_scalar('tag:yaml.org,2002:str', data))
yaml.dump(str(u'你好')) #=> throws UnicodeDecodeError: 'utf8' codec can't decode byte 0xbd in position 0: invalid start byte
```

- (`v1.0.10` or lower) Shell script text was converted to type `<str>`.
  And it was re-converted to literal_unicode by `format_shell_scripts` method.  
  https://github.com/ktdreyer/jenkins-job-wrecker/blob/v1.0.10/jenkins_job_wrecker/cli.py#L108  
  So after all, `shell_element.text` was treated as type `<unicode>`.

- (`v1.0.11` or higher) the `format_shell_scripts` method was removed.
  So `shell_element.text` was treated as type `<str>`, and this occurs `UnicodeDecodeError`.


## How to fix

- In shell builder, treat `shell_element.text` as is.
- Revert unicode representer.

ktdreyer commented:
Thank you so much for writing a test for this!

I just merged a large refactor to master (v1.2.0). Would you mind verifying that this bug is still an issue, and if so, rebase your changes?

(CC'ing @jcderpton in case he wants to review)

jcderpton commented:
Looks fine, merge master in first though to resolve conflicts caused by the refactor.

hell0again commented:
Thanks!

```
- job['name'] = str(name)
+ job['name'] = name
```

I found that non-ascii job name throws error in both `str(name)` or `name`.
The `args.name` is type `str`, not `<unicode>`.
So I converted `name` to `<unicode>`, and also changed `non-ascii` test to give non-ascii job name.

The `non-ascii` test fails on `v1.3.0`, and rebased revision [008a41a](https://github.com/ktdreyer/jenkins-job-wrecker/pull/38/commits/008a41a3f2e65d74edd5ea4236f46c7f7abecb39) would fix the bug.
This revision pass all test in my environment, but fails on travis.

We have to stop using `call()` to run `jenkins_jobs test` in test_jjb.py#L22:
https://github.com/ktdreyer/jenkins-job-wrecker/blob/v1.3.0/tests/test_jjb.py#L22

When running `jenkins_jobs test` with `call()`, jenkins_jobs run with default encoding (i.e. 'ascii').

ktdreyer commented:
My apologies for the delay in merging this. Thank you for contributing the fix

ktdreyer commented:
I've tagged v1.4.0 with this change and pushed it to PyPI. Thank you!
